### ISSUE TITLE
Magento Integration Tests fixtures do not work or are using deprecated methods

### ISSUE THREAD

**Issue by lbajsarowicz** [magento/magento2#27996]
### Preconditions (*)
There are about 300 Product Fixtures for Integration Tests in Magento. I was looking for **at least one** best-practice example of fixture of a Product that `isSalable` returns `true`. Let's search together. Let's review what Magento Fixtures give us.

#### Failures
I won't copy the test result for all the Fixtures, as it's pretty the same:
```bash
$ vendor/bin/phpunit -c $(pwd)/dev/tests/integration/phpunit.xml $(pwd)/dev/tests/integration/testsuite/M2Coach/InventoryFixtures/FixtureTest
PHPUnit 6.5.14 by Sebastian Bergmann and contributors.

F                                                                                                                   1 / 1 (100%)

Time: 4.04 seconds, Memory: 68.50MB

There was 1 failure:

1) M2Coach\InventoryFixtures\FixtureTest::testSalabilityOfFixture
Failed asserting that false is true.

/var/www/html/dev/tests/integration/testsuite/M2Coach/InventoryFixtures/FixtureTest.php:24
/var/www/html/vendor/phpunit/phpunit/phpunit:53

FAILURES!
Tests: 1, Assertions: 1, Failures: 1.

=== Memory Usage System Stats ===
Memory usage (OS):      99.01M (148.88% of 66.50M reported by PHP)
Estimated memory leak:  32.51M (32.83% of used memory)
```

:red_circle:  `Magento/ConfigurableProduct/_files/associated_products.php`
:red_circle: `Magento/Catalog/Model/ResourceModel/_files/few_simple_products.php`

#### Passing, but discouraged
These fixtures use hacks or deprecated methods to make product salable:
```php
    ->setStockData(['use_config_manage_stock' => 1, 'qty' => 100, 'is_in_stock' => 1])
```
:yellow_circle: `Magento/Bundle/_files/bundle_product_checkbox_options.php`
:yellow_circle: `Magento/Bundle/_files/bundle_product_checkbox_required_option.php`
:yellow_circle: `Magento/Bundle/_files/bundle_product_checkbox_required_options.php`
:yellow_circle: `Magento/Bundle/_files/bundle_product_dropdown_options.php`
:yellow_circle: `Magento/Bundle/_files/bundle_product_dropdown_required_options.php`
:yellow_circle: `Magento/Bundle/_files/bundle_product_dropdown_required_options.php`
:yellow_circle: `Magento/Bundle/_files/bundle_product_multiselect_required_option.php`
:yellow_circle: `Magento/Bundle/_files/bundle_product_multiselect_required_options.php`
:yellow_circle: `Magento/Bundle/_files/bundle_product_radio_options.php`
:yellow_circle: `Magento/Bundle/_files/bundle_product_radio_required_option.php`
:yellow_circle: `Magento/Bundle/_files/bundle_product_radio_required_options.php`
:yellow_circle: `Magento/Bundle/_files/bundle_product_with_not_visible_children.php`
:yellow_circle: `Magento/CatalogRule/_files/configurable_product.php`
OK. To be honest. This does not make sense to list them all.

#### Passing, recommended
There is no directly in Magento project, but `magento/inventory` use some considered as correct:
:green_circle: https://github.com/magento/inventory/blob/84c311c8bee8c021e470cce70e18ba4d8517448f/InventorySales/Test/Integration/IsProductSalable/IsSalableWithReservationsConditionTest.php#L96-L118
:green_circle: https://github.com/magento/inventory/blob/84c311c8bee8c021e470cce70e18ba4d8517448f/InventorySales/Test/Integration/Order/PlaceOrderOnDefaultStockTest.php#L97-L120

#### Wrap up:
There is **no best-practices example** for setting up a Product Fixture in Magento Integration Tests.

### Steps to reproduce (*)
The test I am going to use is:

```php
/**
     * @magentoAppIsolation enabled
     * @magentoDbIsolation enabled
     * @magentoDataFixture {PATH_TO_FIXTURE}
     */
    public function testSalabilityOfFixture()
    {
        $objectManager = Bootstrap::getObjectManager();
        $productRepository = $objectManager->get(ProductRepositoryInterface::class);
        $product = $productRepository->get('{PRODUCT_SKU}');
        $this->assertTrue($product->isSalable());
    }
```

Stupid simple.

### Expected result (*)
Tests to the project should be the live example of best-practices and project's know-how.

### Actual result (*)
You should use other, better ways to create fixtures. The best, so far, was @schmengler 's https://github.com/tddwizard/magento2-fixtures

**Comment by lbajsarowicz** [comment]
I've received valuable resources **for Magento with MSI installed**

https://github.com/magento/inventory/blob/84c311c8bee8c021e470cce70e18ba4d8517448f/InventorySales/Test/Integration/IsProductSalable/IsSalableWithReservationsConditionTest.php#L96-L118

https://github.com/magento/inventory/blob/84c311c8bee8c021e470cce70e18ba4d8517448f/InventorySales/Test/Integration/Order/PlaceOrderOnDefaultStockTest.php#L97-L120

Thank you @maghamed for these.

**Comment by lbajsarowicz** [comment]
The issue is still applicable.

**Comment by magento-engcom-team** [comment]
:white_check_mark: Confirmed by @gabrieldagama 
 Thank you for verifying the issue. Based on the provided information internal tickets `MC-40336` were created

**Issue Available**: @gabrieldagama, _You will be automatically unassigned. Contributors/Maintainers can claim this issue to continue. To reclaim and continue work, reassign the ticket to yourself._

**Comment by magento-engcom-team** [comment]
Hi @lbajsarowicz. 

Thank you for your report and collaboration!

The related internal Jira ticket `MC-40336` was closed as `Fixed`. 

The fix will be available with the upcoming `2.4.3` release.
